Automattic/jetpack
# AI Assistant: Rename useSuggestionsFromOpenAI to useAIAssistant and deduplicate suggestion logic

Fixes #36867

## Proposed changes:

* Removes unused function calling code
* Adds `onStop` and `onAllErrors` callbacks and `initialRequestingState` prop to `useAiSuggestions`
* Adds `ERROR_CONTEXT_TOO_LARGE` to the list of errors
* Expose `handleErrorQuotaExceededError` to keep state handling in the hook
* Adds `closeEventSource` to, well, close the event source on Done and on Stop independently
* Removes all suggestion logic from the block's `useSuggestionsFromOpenAI` hook, using the functionality from `useAiSuggestions` instead
* Renames `useSuggestionsFromOpenAI` to `useAIAssistant` as this hook now only has logic that is specific to the block behavior
* Adjusts the AI Assistant code to those changes

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






This one is a bit complex, but the goal is code simplification and deduplication, so there is no new expected behavior. The AI Assistant block, its extensions and other features should work as usual.
 
* Go to the block editor
* Add AI Assistant blocks and generate content
* Regenerate content, ask for changes, apply toolbar options, force an error by simulating being offline
* Check that nothing breaks and things work as usual
* Check the console. Duplicated and broken debug messages should not be present anymore
* Add a Form block and test it as well
* Check the other features (excerpt, feedback, etc)
